crates/bevy_reflect/src/func/mod.rs

```diff
@@ -34,22 +34,81 @@
 //! assert_eq!(value.unwrap_owned().downcast_ref::<i32>(), Some(&100));
 //! ```
 //!
+//! # Function vs Closure
+//!
+//! In Rust, a "function" is any callable that does not capture its environment.
+//! These are typically defined with the `fn` keyword, but may also use anonymous function syntax.
```

Comment:
Demonstrating what the "anonymous function syntax" is will be useful to readers here.